single-pool/js/src/transactions.ts

```diff
@@ -0,0 +1,302 @@
+        pool,
+        userStakeAccount as PublicKey,
+        userTokenAccount,
+        userLamportAccount,
+      ),
+    );
+
+    return transaction;
+  }
+
+  static async withdraw(params: WithdrawParams) {
+    const { connection, pool, userWallet, userStakeAccount, tokenAmount, createStakeAccount } =
+      params;
+
+    if (typeof tokenAmount == 'bigint' && tokenAmount > BigInt(Number.MAX_SAFE_INTEGER)) {
+      throw 'cannot convert tokenAmount to Number';
+    }
+
+    const transaction = new Transaction();
+
+    const poolMintAuthority = findPoolMintAuthorityAddress(this.programId, pool);
+
+    const userStakeAuthority = params.userStakeAuthority || userWallet;
+    const userTokenAccount =
+      params.userTokenAccount ||
+      getAssociatedTokenAddressSync(findPoolMintAddress(this.programId, pool), userWallet);
+    const userTokenAuthority = params.userTokenAuthority || userWallet;
+
+    if (createStakeAccount) {
+      transaction.add(
+        SystemProgram.createAccount({
+          fromPubkey: userWallet,
+          lamports: await connection.getMinimumBalanceForRentExemption(StakeProgram.space),
+          newAccountPubkey: userStakeAccount,
+          programId: StakeProgram.programId,
+          space: StakeProgram.space,
+        }),
+      );
+    }
+
+    transaction.add(
+      createApproveInstruction(
+        userTokenAccount,
+        poolMintAuthority,
+        userTokenAuthority,
+        Number(tokenAmount),
+      ),
+    );
+
+    transaction.add(
+      SinglePoolInstruction.withdrawStake(
+        pool,
+        userStakeAccount,
+        userStakeAuthority,
+        userTokenAccount,
+        userTokenAuthority,
+        tokenAmount,
+      ),
+    );
+
+    return transaction;
+  }
+
+  static createTokenMetadata(pool: PublicKey, payer: PublicKey) {
+    const transaction = new Transaction();
+    transaction.add(SinglePoolInstruction.createTokenMetadata(pool, payer));
+
+    return transaction;
+  }
+
+  static updateTokenMetadata(
+    voteAccount: PublicKey,
+    authorizedWithdrawer: PublicKey,
+    name: string,
+    symbol: string,
+    uri?: string,
+  ) {
+    const transaction = new Transaction();
+    transaction.add(
+      SinglePoolInstruction.updateTokenMetadata(
+        voteAccount,
+        authorizedWithdrawer,
+        name,
+        symbol,
+        uri,
+      ),
+    );
+
+    return transaction;
+  }
+
+  static async createAndDelegateUserStake(
+    connection: Connection,
+    voteAccount: PublicKey,
+    userWallet: PublicKey,
+    stakeAmount: number | bigint,
+  ) {
+    const transaction = new Transaction();
+
+    const pool = findPoolAddress(this.programId, voteAccount);
+    const stakeAccount = await findDefaultDepositAccountAddress(pool, userWallet);
+
+    const seed = 'svsp' + pool.toString().slice(0, 28);
+    const stakeRent = await connection.getMinimumBalanceForRentExemption(StakeProgram.space);
+
+    if (
+      typeof stakeAmount == 'bigint' &&
```

Comment:
Again, shouldn't this be?
```suggestion
      typeof stakeAmount != 'bigint' &&
```